src/Parser/Hunter/Marksmanship/Modules/Spells/Trueshot.js

```diff
@@ -23,23 +24,46 @@ class Trueshot extends Analyzer {
   totalCritsInTS = 0;
   totalCastsPrTS = 0;
   executeTrueshots = 0;
+  startFocusForCombatant = 0;
+  prepullTrueshots = 0;
 
-  on_byPlayer_cast(event) {
+  on_byPlayer_applybuff(event) {
     const buffId = event.ability.guid;
-    if (buffId !== SPELLS.TRUESHOT.id) {
+    if (buffId !== SPELLS.TRUESHOT.id && !event.prepull) {
       return;
     }
-    this.trueshotCasts += 1;
-    this.accumulatedFocusAtTSCast += event.classResources[0]['amount'] || 0;
-    if (this.combatants.selected.hasBuff(SPELLS.BULLSEYE_TRAIT.id, event.timestamp)) {
-      this.executeTrueshots += 1;
+    //ensures we only get 1 of these events
+    if (this.trueshotCasts === 0) {
```

Comment:
I'm unclear on how you could ever get more than one, is this a 'just in case' kind of thing?

Reply:
It registered more than one if I didn't do it, if I recall correctly, but otherwise, ye just a "just in case"

Reply:
It registered more than one because your early return statement is incorrect, as I noted above.